backend/src/lib/report-generator.js

```diff
@@ -166,19 +324,85 @@ expressions.filters.NewLines = function(input) {
     return result;
 }
 
+// Embeds input within OOXML paragraph tags, applying an optional style name to it: {@input | p: 'Some style'}
+expressions.filters.p = function(input, style = null) {
+    let result = '<w:p>';
+
+    if (style !== null ) {
+        let style_parsed = style.replaceAll(' ', '');
+        result += '<w:pPr><w:pStyle w:val="' + style_parsed + '"/></w:pPr>';
+    }
+    result += input + '</w:p>';
+
+    return result;
+}
+
+// Reverses the input array: {input | reverse}
+expressions.filters.reverse = function(input) {
+    return input.reverse();
+}
+
+// Looks up an attribute from a sequence of objects, doted notation is supported: {findings | select: 'cvss.environmentalSeverity'}
+expressions.filters.select = function(input, attr) {
+    return input.map(function(item) { return _getPropertyValue(item, attr) });
+}
+
+// Sorts the input array according an optional given attribute, dotted notation is supported: {#findings | sort 'cvss.environmentalSeverity'}{name}{/findings | sort 'cvss.environmentalSeverity'}
+expressions.filters.sort = function(input, key = null) {
+    if (key === null) {
+        return input.sort();
+    }
+    else {
+        return input.sort(function(a, b) {
+            return _getPropertyValue(a, key) < _getPropertyValue(b, key);
+        });
+    }
+}
+
 // Sort array by supplied field: {#findings | sortArrayByField: 'identifier':1}{/}
 // order: 1 = ascending, -1 = descending
 expressions.filters.sortArrayByField = function (input, field, order) {
     //invalid order sort ascending
     if(order != 1 && order != -1) order = 1;
-    
+
     const sorted = input.sort((a,b) => {
         //multiply by order so that if is descending (-1) will reverse the values
         return _.get(a, field).localeCompare(_.get(b, field), undefined, {numeric: true}) * order
-    })    
+    })
     return sorted;
 }
 
+// Takes a string as input and split it into an ordered list using a separator: {input | split: ', '}
+expressions.filters.split = function(input, sep) {
```

Comment:
Yes, I added it as the counterpart of the `split` filter, however this may indeed not be very useful in the case of Pwndoc so I removed it in [d24b10b](https://github.com/pwndoc/pwndoc/pull/281/commits/d24b10b93e5d3fc8a8823e188edafd28d1f8138e).